tests/unit/storage_v2_analytics_mode.cpp

```diff
@@ -0,0 +1,71 @@
+#include <gtest/gtest.h>
+#include <string_view>
+
+#include "storage/v2/isolation_level.hpp"
+#include "storage/v2/storage.hpp"
+#include "storage/v2/vertex_accessor.hpp"
+
+namespace {
+
+int64_t VerticesCount(memgraph::storage::Storage::Accessor &accessor) {
+  int64_t count{0};
+  for ([[maybe_unused]] const auto &vertex : accessor.Vertices(memgraph::storage::View::OLD)) {
+    ++count;
+  }
+  return count;
+}
+
+inline constexpr std::array storage_modes{
+    memgraph::storage::StorageMode::IN_MEMORY_ANALYTICAL,
+    memgraph::storage::StorageMode::IN_MEMORY_TRANSACTIONAL,
+};
+
+}  // namespace
+
+std::string_view StorageModeToString(memgraph::storage::StorageMode storage_mode) {
+  switch (storage_mode) {
+    case memgraph::storage::StorageMode::IN_MEMORY_ANALYTICAL:
+      return "IN_MEMORY_ANALYTICAL";
+    case memgraph::storage::StorageMode::IN_MEMORY_TRANSACTIONAL:
+      return "IN_MEMORY_TRANSACTIONAL";
+  }
+}
+
+class StorageModeTest : public ::testing::TestWithParam<memgraph::storage::StorageMode> {
+ public:
+  struct PrintStringParamToName {
+    std::string operator()(const testing::TestParamInfo<memgraph::storage::StorageMode> &info) {
+      return std::string(StorageModeToString(static_cast<memgraph::storage::StorageMode>(info.param)));
+    }
+  };
+};
+
+// you should be able to see nodes if there is analytics mode
+TEST_P(StorageModeTest, Mode) {
+  const memgraph::storage::StorageMode storage_mode = GetParam();
+
+  memgraph::storage::Storage storage{
+      {.transaction{.isolation_level = memgraph::storage::IsolationLevel::SNAPSHOT_ISOLATION}}};
```

Comment:
I think this test is good. What do you think about simple extension that you also test read_committed? I think logically is not important but maybe as an idea

Reply:
I think it is a bonus, as it doesn't have any impact. Because if it works on snapshot isolation, should also work on other states. I would leave it as it is now. What do you think?

Reply:
Your call, I agree that it shouldn't create problems.